src/datahike/writer.cljc

```diff
@@ -40,44 +40,49 @@
     [transaction-queue commit-queue
      (thread-try
       S
-      (let [store (:store @(:wrapped-atom connection))]
+      (do
         ;; processing loop
-        (go-loop []
-          (if-let [{:keys [op args callback] :as invocation} (<?- transaction-queue)]
-            (do
-              (when (> (count transaction-queue-buffer) (* 0.9 transaction-queue-size))
-                (log/warn "Transaction queue buffer more than 90% full, "
-                          (count transaction-queue-buffer) "of" transaction-queue-size  " filled."
-                          "Reduce transaction frequency."))
-              (let [op-fn (write-fn-map op)
-                    res   (try
-                            (apply op-fn connection args)
+        (go
+         ;; delay processing until the writer we are part of in connection is set
+          (<! (timeout 10))
```

Comment:
This is a bit ugly, it is necessary because of a circular dependency on the writer which this loop is part of. This waits until the writer is constructed on the caller (which happens immediately after this function returns.